packages/knip/src/typescript/resolveModuleNames.ts

```diff
@@ -147,3 +155,19 @@ export function createCustomModuleResolver(
 
   return timerify(resolveModuleNames);
 }
+
+const declarationPathRe = /^(.*)\.d(\.[^.]+)\.ts$/;
+
+/**
+ * For paths that look like `.../foo.d.yyy.ts` returns path `.../foo.yyy` and
```

Comment:
Trying to use "foo" as little as possible myself, any chance you could rid of that too? I guess "module" could do.